membership-common/src/main/scala/com/gu/memsub/subsv2/reads/SubPlanReads.scala

```diff
@@ -1,112 +1,40 @@
-  private val weeklyRestOfWorldReads: SubPlanReads[WeeklyRestOfWorld] = findProduct(_.weeklyRestOfWorld.point[List], WeeklyRestOfWorld)
-
-  implicit val productReads: SubPlanReads[Product] = new SubPlanReads[Product] {
-    override def read(p: ProductIds, z: SubscriptionZuoraPlan, c: CatalogZuoraPlan): ValidationNel[String, Product] =
-      (contentSubscriptionReads.read(p, z, c) orElse2
-        membershipReads.read(p, z, c) orElse2
-        contributionReads.read(p, z, c)).withTrace("productReads")
-  }
-
-  private val contentSubscriptionReads: SubPlanReads[ContentSubscription] = new SubPlanReads[ContentSubscription] {
-    override def read(p: ProductIds, z: SubscriptionZuoraPlan, c: CatalogZuoraPlan): ValidationNel[String, ContentSubscription] =
-      (paperReads.read(p, z, c) orElse2
-        zDigipackReads.read(p, z, c) orElse2
-        supporterPlusReads.read(p, z, c)).withTrace("contentSubscriptionReads")
-  }
-
-  private val paperReads: SubPlanReads[Paper] = new SubPlanReads[Paper] {
-    override def read(p: ProductIds, z: SubscriptionZuoraPlan, c: CatalogZuoraPlan): ValidationNel[String, Paper] =
-      (voucherReads.read(p, z, c).map(identity[Paper]) orElse2
-        digitalVoucherReads.read(p, z, c) orElse2
-        deliveryReads.read(p, z, c) orElse2
-        nationalDeliveryReads.read(p, z, c) orElse2
-        weeklyZoneAReads.read(p, z, c) orElse2
-        weeklyZoneBReads.read(p, z, c) orElse2
-        weeklyZoneCReads.read(p, z, c) orElse2
-        weeklyDomesticReads.read(p, z, c) orElse2
-        weeklyRestOfWorldReads.read(p, z, c)).withTrace("paperReads")
-  }
-
-  implicit def anyPlanReads[P <: Product, C <: ChargeList](implicit
-      productReads: SubPlanReads[P],
-      chargeListReads: ChargeListReads[C],
-  ): SubPlanReads[SubscriptionPlan[P, C]] =
-    new SubPlanReads[SubscriptionPlan[P, C]] {
-      override def read(
-          ids: ProductIds,
-          subZuoraPlan: SubscriptionZuoraPlan,
-          catZuoraPlan: CatalogZuoraPlan,
-      ): ValidationNel[String, SubscriptionPlan[P, C]] =
-        (for {
-          charges <- chargeListReads.read(catZuoraPlan.benefits, subZuoraPlan.charges.list.toList)
-          product <- productReads.read(ids, subZuoraPlan, catZuoraPlan)
-        } yield {
-          val highLevelFeatures = subZuoraPlan.features.map(com.gu.memsub.Subscription.Feature.fromRest)
-          SubscriptionPlan(
-            subZuoraPlan.id,
-            catZuoraPlan.id,
-            catZuoraPlan.name,
-            catZuoraPlan.description,
-            subZuoraPlan.productName,
-            subZuoraPlan.lastChangeType,
-            catZuoraPlan.productType,
-            product,
-            highLevelFeatures,
-            charges,
-            subZuoraPlan.chargedThroughDate,
-            subZuoraPlan.start,
-            subZuoraPlan.end,
-          )
-        }).withTrace("paidPlanReads")
-    }
+  def anyPlanReads(
```

Comment:
```suggestion
  def planReads(
```

Seeing the term `anyPlan` in the code gives me PTSD

Reply:
done.  I did think about moving it to SubscriptionTransform as that's the only place it's used, but I SubPlanReads.scala is a familiar file